src/side-navigation/internal.tsx

```diff
@@ -64,63 +64,147 @@ export function Header({ definition, activeHref, fireFollow }: HeaderProps) {
+            </li>
+          ),
+        });
+        return;
+      }
+      case 'link-group': {
+        lists[currentListIndex].items?.push({
+          element: (
+            <li key={index} data-itemid={`item-${itemid}`} className={styles['list-item']}>
+              <LinkGroup definition={item} activeHref={activeHref} fireChange={fireChange} fireFollow={fireFollow} />
+            </li>
+          ),
+        });
+        return;
+      }
+      case 'expandable-link-group': {
+        lists[currentListIndex].items?.push({
+          element: (
+            <li key={index} data-itemid={`item-${itemid}`} className={styles['list-item']}>
+              <ExpandableLinkGroup
+                definition={item}
+                activeHref={activeHref}
+                fireChange={fireChange}
+                fireFollow={fireFollow}
+                variant={variant}
+              />
+            </li>
+          ),
+        });
+        return;
+      }
+    }
+  });
+
   return (
-    <ul className={clsx(styles.list, styles[`list-variant-${variant}`])}>
-      {items.map((item, i) => (
-        <li key={i} className={styles['list-item']}>
-          {item.type === 'link' && (
-            <Link definition={item} activeHref={activeHref} fireChange={fireChange} fireFollow={fireFollow} />
-          )}
-          {item.type === 'section' && (
-            <Section
-              definition={item}
-              activeHref={activeHref}
-              fireChange={fireChange}
-              fireFollow={fireFollow}
-              variant={variant}
-            />
-          )}
-          {item.type === 'section-group' && (
-            <SectionGroup definition={item} activeHref={activeHref} fireChange={fireChange} fireFollow={fireFollow} />
-          )}
-          {item.type === 'link-group' && (
-            <LinkGroup definition={item} activeHref={activeHref} fireChange={fireChange} fireFollow={fireFollow} />
-          )}
-          {item.type === 'expandable-link-group' && (
-            <ExpandableLinkGroup
-              definition={item}
-              activeHref={activeHref}
-              fireChange={fireChange}
-              fireFollow={fireFollow}
-              variant={variant}
-            />
-          )}
-          {((i === 0 && item.type === 'divider') || (items[i + 1] && items[i + 1].type === 'divider')) && (
-            <Divider variant="default" />
-          )}
-        </li>
-      ))}
-    </ul>
+    <>
+      {lists.map((list, index) => {
+        if (!list.items || list.items.length === 0) {
+          return (
+            <div key={`hr-${index}`} className={clsx(styles.list, styles[`list-variant-${variant}`])}>
+              {list.element}
+            </div>
+          );
+        } else {
+          return (
+            <ul key={`list-${index}`} className={clsx(styles.list, styles[`list-variant-${list.listVariant}`])}>
```

Comment:
I'm not sure if multiple lists after one another are the correct semantic structure here. Maybe nested lists instead? Alternatively, I'm wondering if `menu` would be even better for side navigation, see for example: https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-editor/.

Reply:
Nested lists won't work, as it'll result in the same problem we tried to solve (`<ul>` that has non-`li` direct children).

Discussed the overall approach with A11y team, they approved it